ScoreSaber/Core/Services/PlayerService.cs

```diff
@@ -33,162 +37,101 @@ public void ChangeLoginStatus(LoginStatus _loginStatus, string status) {
             ChangeLoginStatus(LoginStatus.InProgress, "Signing into ScoreSaber...");
-
             var platformUserModel = Plugin.Container.TryResolve<IPlatformUserModel>();
-            var authToken = await platformUserModel.GetUserAuthToken();
             var userInfo = await platformUserModel.GetUserInfo(CancellationToken.None);
-
-            var nonce = string.Empty;
-            var platform = string.Empty;
-
-            switch (userInfo.platform) {
-                case UserInfo.Platform.Steam:
-                    nonce = authToken.token;
-                    platform = "0";
-                    break;
-                case UserInfo.Platform.Oculus:
-                    nonce = authToken.token + "," + (await platformUserModel.RequestXPlatformAccessToken(CancellationToken.None)).token;
-                    platform = "1";
-                    break;
+            var authToken = await platformUserModel.GetUserAuthToken();
+            var (nonce, platform) = userInfo.platform switch {
+                UserInfo.Platform.Steam => (authToken.token, "0"),
+                UserInfo.Platform.Oculus => ($"{authToken.token},{(await platformUserModel.RequestXPlatformAccessToken(CancellationToken.None)).token}", "1"),
+                _ => (string.Empty, string.Empty)
+            };
+            if (string.IsNullOrEmpty(nonce) || string.IsNullOrEmpty(platform)) {
+                ChangeLoginStatus(LoginStatus.Error, "Failed to authenticate with ScoreSaber: Invalid platform");
+                return;
             }
-
-            var playerId = userInfo.platformUserId;
-            var playerName = userInfo.userName;
             var friendIds = await platformUserModel.GetUserFriendsUserIds(false);
-            var friends = string.Join(",", friendIds.Where(x => x != "0"));
-
-            var playerInfo = new LocalPlayerInfo(playerId, playerName, friends, platform, nonce);
-
-            int attempts = 1;
-
-            while (attempts < 4) {
-
-                var authenticated = await AuthenticateWithScoreSaber(playerInfo);
-
-                if (authenticated) {
+            var playerInfo = new LocalPlayerInfo(
+                userInfo.platformUserId,
+                userInfo.userName,
+                string.Join(",", friendIds.Where(x => x != "0")),
+                platform,
+                nonce
+            );
+            for (int attempt = 1; attempt <= 3; attempt++) {
+                if (await AuthenticateWithScoreSaber(playerInfo)) {
                     localPlayerInfo = playerInfo;
-                    string successText = "Successfully signed into ScoreSaber!";
-                    if (localPlayerInfo.playerId == PlayerIDs.Denyah)
-                        successText = "Wagwan piffting wots ur bbm pin?";
-
+                    var successText = localPlayerInfo.playerId == PlayerIDs.Denyah
+                        ? "Wagwan piffting wots ur bbm pin?"
+                        : "Successfully signed into ScoreSaber!";
```

Comment:
Please don't overuse ternery operators just because you can. Often the version without is much more readable (and this is one of those cases).

Especially in a case like this where one is the default for everyone and one of them is the exceptional case.

Reply:
fixed